cors/resources/cors-makeheader.py

```diff
@@ -3,7 +3,7 @@
 from wptserve.utils import isomorphic_decode
 
 def main(request, response):
-    origin = request.GET.first(b"origin", request.headers.get(b'origin'))
+    origin = request.GET.first(b"origin", request.headers.get(b'origin') or b'none')
```

Comment:
When would we need the fallback here? Why doesn't the second request include the origin header?

Reply:
Because same-origin requests doesn't have an origin header.

Reply:
Rephrasing:

> When would we need the fallback here?

The "origin: echo" test where _sameorigin_ is true.

> Why doesn't the second request include the origin header?

Do you mean the origin component in the request URL query?
I think the test intention is:

- The request fails when access-control-allow-origin header is missing except for the same origin case (the first test case).
- The request succeeds when access-control-allow-origin header is correctly set (the second test case).

So, I think attaching "&origin=${document.origin}" to the URL of the second request would fix the failure without changing the python script. Do you prefer the solution?

Reply:
I guess slightly, but I'm okay with this too given it's an old test.